File: leetcode/easy/1528/1528.rb
# Runtime: 68 ms, faster than 41.18%
# Memory Usage: 210.4 MB, less than 36.97%
def restore_string(s, indices)
    result = 'a' * indices.size
    j = 0
    indices.each do |i|
        result[i] = s[j] 
        j += 1
    end
    
    return result
end

File: leetcode/easy/1528/1528.cpp
/**
 * Runtime: 8 ms, faster than 94.80% 
 * Memory Usage: 15.4 MB, less than 63.23%
 */
class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        string result (indices.size(), 'a');
        int j=0;
        for(int i : indices) 
        {
            result[i] = s[j++];
        }
        
        return result;
    }
};
